plugins/org.jkiss.dbeaver.model.lsm/grammar/SQLStandardParser.g4

```diff
@@ -468,18 +400,16 @@ selectStatementSingleRow: SELECT (setQuantifier)? selectList INTO selectTargetLi
 selectTargetList: targetSpecification ((Comma targetSpecification)+)?;
 
 // data change statements
-sqlDataChangeStatement: (deleteStatementPositioned|deleteStatementSearched|insertStatement|updateStatementPositioned|updateStatementSearched);
-deleteStatementPositioned: DELETE FROM tableName WHERE CURRENT OF cursorName;
-deleteStatementSearched: DELETE FROM tableName (WHERE searchCondition)?;
+sqlDataChangeStatement: (deleteStatement||insertStatement|updateStatement);
+deleteStatement: DELETE FROM tableName (WHERE CURRENT OF cursorName)|(WHERE searchCondition)?;
 insertStatement: INSERT INTO tableName insertColumnsAndSource;
 insertColumnsAndSource: ((LeftParen insertColumnList RightParen)? queryExpression|DEFAULT VALUES);
 insertColumnList: columnNameList;
-updateStatementPositioned: UPDATE tableName SET setClauseList WHERE CURRENT OF cursorName;
+updateStatement: UPDATE tableName SET setClauseList (WHERE CURRENT OF cursorName)|(WHERE searchCondition)?;
```

Comment:
Maybe it would be better to rewrite it as:

```suggestion
updateStatement: UPDATE tableName SET setClauseList (WHERE (CURRENT OF cursorName)|searchCondition)?;
```